tests/test_discriminated_union.py

```diff
@@ -942,10 +942,10 @@ def test_lax_or_strict_definitions() -> None:
     )
     discriminated_schema = apply_discriminator(core_schema.union_schema(cat, dog), 'kind')
     assert discriminated_schema == {
-        'choices': {
-            'DOG': {
-                'definitions': [{'ref': 'my-str-definition', 'type': 'str'}],
-                'schema': {
+        'definitions': [{'ref': 'my-str-definition', 'type': 'str'}],
```

Comment:
note, applying the discriminated union moves the definitions around a bit here, but the end result is still valid. I think I am implicitly making the assumption that definitions schemas can't themselves have a 'ref'